topic-operator/src/main/java/io/strimzi/operator/topic/TopicOperator.java

```diff
@@ -357,16 +372,75 @@ public TopicOperator(Vertx vertx, Kafka kafka,
                          TopicStore topicStore,
                          Labels labels,
                          String namespace,
-                         Config config) {
+                         Config config,
+                         MetricsProvider metrics) {
         this.kafka = kafka;
         this.k8s = k8s;
         this.vertx = vertx;
         this.labels = labels;
         this.topicStore = topicStore;
         this.namespace = namespace;
         this.config = config;
+        this.metrics = metrics;
+
+        initMetrics();
+    }
+
+    public void initMetrics() {
+        Tags metricTags = Tags.of(Tag.of("kind", "KafkaTopic"));
+
+        periodicReconciliationsCounter = metrics.counter(METRICS_PREFIX + "reconciliations.periodical",
+                "Number of periodical reconciliations done by the operator",
+                metricTags);
+
+        reconciliationsCounter = metrics.counter(METRICS_PREFIX + "reconciliations",
+                "Number of reconciliations done by the operator for individual topics",
+                metricTags);
+
+        failedReconciliationsCounter = metrics.counter(METRICS_PREFIX + "reconciliations.failed",
+                "Number of reconciliations done by the operator for individual topics which failed",
+                metricTags);
+
+        successfulReconciliationsCounter = metrics.counter(METRICS_PREFIX + "reconciliations.successful",
+                "Number of reconciliations done by the operator for individual topics which were successful",
+                metricTags);
+
+        topicCounter = metrics.gauge(METRICS_PREFIX + "resources",
+                "Number of topics the operator sees",
+                metricTags);
+
+        reconciliationsTimer = metrics.timer(METRICS_PREFIX + "reconciliations.duration",
+                "The time the reconciliation takes to complete",
```

Comment:
Is this an average? Or just the time for the last reconciliation? We probably need two of these because of the fact that periodic reconciliations (because they reconcile all the topics) will take ~n times longer than event-based reconciliations.

Reply:
@scholzj Did you distinguished these?

Reply:
That is handled by the Micrometer framework it self. I would need to check what exactly it provides and what we used. I believe it might provide the max time and the average?

Reply:
OK, I see. It just seems to measure the time for periodic reconciliations (as Jakub mentioned [here](https://github.com/strimzi/strimzi-kafka-operator/pull/2874/files#r413344754)), and doesn't cover the reconciliations caused by changes. So the number will make sense, but it also means we have no visibility of the time taken to perform per-topic reconciliation.